tests/integration/test_sns.py

```diff
@@ -1045,6 +1045,166 @@ def get_messages(queue_url):
+                    assert message["MessageAttributes"]["attr1"] == {
+                        "StringValue": "19.12",
+                        "DataType": "Number",
+                    }
+
+                elif message["Body"] == "Test Message without attribute":
+                    assert message.get("MessageAttributes") is None
+
+        retry(get_messages, retries=5, sleep=1, queue_url=queue_url)
+
+    def test_publish_batch_exceptions(self):
+        topic_name = "topic-{}.fifo".format(short_uid())
+        queue_name = "queue-%s.fifo" % short_uid()
+
+        topic_arn = self.sns_client.create_topic(Name=topic_name, Attributes={"FifoTopic": "true"})[
+            "TopicArn"
+        ]
+        queue = self.sqs_client.create_queue(
+            QueueName=queue_name,
+            Attributes={"FifoQueue": "true"},
+        )
+
+        queue_url = queue["QueueUrl"]
+        queue_arn = aws_stack.sqs_queue_arn(queue_name)
+
+        self.sns_client.subscribe(
+            TopicArn=topic_arn,
+            Protocol="sqs",
+            Endpoint=queue_arn,
+            Attributes={"RawMessageDelivery": "true"},
+        )
+
+        with pytest.raises(ClientError) as e:
+            self.sns_client.publish_batch(
+                TopicArn=topic_arn,
+                PublishBatchRequestEntries=[
+                    {
+                        "Id": "1",
+                        "Message": "Test Message with two attributes",
+                    }
+                ],
+            )
+        assert e.value.response["Error"]["Code"] == "InvalidParameter"
+        assert (
+            e.value.response["Error"]["Message"]
+            == "The MessageGroupId parameter is required for FIFO topics"
+        )
+        assert e.value.response["ResponseMetadata"]["HTTPStatusCode"] == 400
+
+        with pytest.raises(ClientError) as e:
+            self.sns_client.publish_batch(
+                TopicArn=topic_arn,
+                PublishBatchRequestEntries=[
+                    {"Id": f"Id_{i}", "Message": f"message_{i}"} for i in range(11)
+                ],
+            )
+        assert e.value.response["Error"]["Code"] == "TooManyEntriesInBatchRequest"
+        assert (
+            e.value.response["Error"]["Message"]
+            == "The batch request contains more entries than permissible"
+        )
+        assert e.value.response["ResponseMetadata"]["HTTPStatusCode"] == 400
+
+        with pytest.raises(ClientError) as e:
+            self.sns_client.publish_batch(
+                TopicArn=topic_arn,
+                PublishBatchRequestEntries=[
+                    {"Id": "1", "Message": f"message_{i}"} for i in range(2)
+                ],
+            )
+        assert e.value.response["Error"]["Code"] == "BatchEntryIdsNotDistinct"
+        assert (
+            e.value.response["Error"]["Message"]
+            == "Two or more batch entries in the request have the same Id"
+        )
```

Comment:
checking for exact error messages might be a bit brittle in the long run, it is probably enough to check for the error code in this and similar cases